apps/studio/src/features/editing-experience/components/TipTapComponent.tsx

```diff
@@ -0,0 +1,185 @@
+import { Box, Flex, Icon, VStack, Text as ChakraText } from '@chakra-ui/react'
+import { Button, IconButton } from '@opengovsg/design-system-react'
+import { Blockquote } from '@tiptap/extension-blockquote'
+import { Bold } from '@tiptap/extension-bold'
+import { BulletList } from '@tiptap/extension-bullet-list'
+import { Document } from '@tiptap/extension-document'
+import { Dropcursor } from '@tiptap/extension-dropcursor'
+import { Gapcursor } from '@tiptap/extension-gapcursor'
+import { Heading } from '@tiptap/extension-heading'
+import { History } from '@tiptap/extension-history'
+import { HorizontalRule } from '@tiptap/extension-horizontal-rule'
+import { Italic } from '@tiptap/extension-italic'
+import { ListItem } from '@tiptap/extension-list-item'
+import { OrderedList } from '@tiptap/extension-ordered-list'
+import { Paragraph } from '@tiptap/extension-paragraph'
+import { Strike } from '@tiptap/extension-strike'
+import { Subscript } from '@tiptap/extension-subscript'
+import { Superscript } from '@tiptap/extension-superscript'
+import { Text } from '@tiptap/extension-text'
+import TableCell from '@tiptap/extension-table-cell'
+import TableHeader from '@tiptap/extension-table-header'
+import TableRow from '@tiptap/extension-table-row'
+import { EditorContent, useEditor } from '@tiptap/react'
+// import type { CustomRendererProps } from './types'
+import { BiImage, BiText, BiX } from 'react-icons/bi'
+import Underline from '@tiptap/extension-underline'
+import { MenuBar } from '~/components/PageEditor/MenuBar'
+
+import { Table } from './extensions/Table'
+
+type NativeComponentType = 'paragraph' | 'image'
+
+export interface TipTapComponentProps {
+  type: NativeComponentType
+  data: any
+  handleChange(path: string, value: any): void
+  onProceed(path: string, value: any): void
+  onClose(): void
+  path: string
+}
+
+const typeMapping = {
+  paragraph: {
+    icon: BiText,
+    title: 'Paragraph',
+  },
+  image: {
+    icon: BiImage,
+    title: 'Image',
+  },
+}
+
+function TipTapComponent({
```

Comment:
this looks vaguely problematic, could we try to add/exit out of the tiptap component? my main concern is that we're re-initialising the editor on every `mount/unmount`, which might cause issues.

Reply:
ah you're right, the existing edits disappear if we exit out of the drawer (e.g. if we want to implement minimizing of the entire drawer, then minimizing the drawer and returning will not preserve the state at current)

I think this might be tackled when we link up the preview and the editor, since this is likely something that should be  handled using localstorage/context?

Reply:
Modified to use context upstream in https://github.com/opengovsg/isomer/pull/219